fsmonitor.c

```diff
@@ -32,17 +46,27 @@ int read_fsmonitor_extension(struct index_state *istate, const void *data,
 	uint32_t ewah_size;
 	struct ewah_bitmap *fsmonitor_dirty;
 	int ret;
+	uint64_t tm;
+	struct strbuf last_update = STRBUF_INIT;
 
 	if (sz < sizeof(uint32_t) + sizeof(uint64_t) + sizeof(uint32_t))
 		return error("corrupt fsmonitor extension (too short)");
 
 	hdr_version = get_be32(index);
 	index += sizeof(uint32_t);
-	if (hdr_version != INDEX_EXTENSION_VERSION)
+	if (hdr_version == INDEX_EXTENSION_VERSION1) {
+		tm = get_be64(index);
+		strbuf_addf(&last_update, "%"PRIu64"", tm);
+		index += sizeof(uint64_t);
+	} else if (hdr_version == INDEX_EXTENSION_VERSION2) {
+		strbuf_addstr(&last_update, index);
```

Comment:
IIRC, `index` is now pointing at the `token` which is a null-terminated string, right?  There's an implicit assumption that the `last_update.len + 1` is at least as big as `sizeof(uint64_t)` because of the guard at the top.

Reply:
Yeah I think that guard should be updated since the token could in theory be less than `sizeof(uint64_t)`.